hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -618,150 +500,143 @@ class ServiceBackendSocketAPI2(
+    val remoteTmpDir = readString()
 
-      case IMPORT_FAM =>
-        val tmpdir = readString()
-        val billingProject = readString()
-        val remoteTmpDir = readString()
-        val path = readString()
-        val quantPheno = readBool()
-        val delimiter = readString()
-        val missing = readString()
-        try {
-          val result = backend.importFam(tmpdir, sessionId, billingProject, remoteTmpDir, path, quantPheno, delimiter, missing, flags)
-          writeBool(true)
-          writeString(result)
-        } catch {
-          case t: Throwable =>
-            writeBool(false)
-            writeString(formatException(t))
+    def withExecuteContext(methodName: String, method: ExecuteContext => String): String = ExecutionTimer.logTime(methodName) { timer =>
+      val fs = retryTransientErrors {
+        using(new FileInputStream(s"${backend.scratchDir}/secrets/gsa-key/key.json")) { is =>
+          new GoogleStorageFS(Some(IOUtils.toString(is, Charset.defaultCharset().toString()))).asCacheable()
         }
+      }
+      ExecuteContext.scoped(
+        tmpdir,
+        "file:///tmp",
+        backend,
+        fs,
+        timer,
+        null,
+        backend.theHailClassLoader,
+        HailFeatureFlags.fromMap(flags)
+      ) { ctx =>
+        ctx.backendContext = new ServiceBackendContext(sessionId, billingProject, remoteTmpDir)
+        method(ctx)
+      }
+    }
 
-      case GOODBYE =>
-        writeInt(GOODBYE)
+    try {
+      val result = (cmd: @switch) match {
+        case LOAD_REFERENCES_FROM_DATASET =>
+          val path = readString()
+          withExecuteContext(
+            "ServiceBackend.loadReferencesFromDataset",
+            backend.loadReferencesFromDataset(_, path)
+          )
+        case VALUE_TYPE =>
+          val s = readString()
+          withExecuteContext(
+            "ServiceBackend.valueType",
+            backend.valueType(_, s)
+          )
+        case TABLE_TYPE =>
+          val s = readString()
+          withExecuteContext(
+            "ServiceBackend.tableType",
+            backend.tableType(_, s)
+          )
+        case MATRIX_TABLE_TYPE =>
+          val s = readString()
+          withExecuteContext(
+            "ServiceBackend.matrixTableType",
+            backend.matrixTableType(_, s)
+          )
+        case BLOCK_MATRIX_TYPE =>
+          val s = readString()
+          withExecuteContext(
+            "ServiceBackend.blockMatrixType",
+            backend.blockMatrixType(_, s)
+          )
+        case REFERENCE_GENOME =>
+          val name = readString()
+          withExecuteContext(
+            "ServiceBackend.referenceGenome",
+            backend.referenceGenome(_, name)
+          )
+        case EXECUTE =>
+          val code = readString()
+          val token = readString()
+          withExecuteContext("ServiceBackend.execute", { ctx =>
+            backend.execute(ctx, code, token)
+          })
```

Comment:
nit: inconsistent with the way all the other lambdas are written

Reply:
missed this message but its now fixed too